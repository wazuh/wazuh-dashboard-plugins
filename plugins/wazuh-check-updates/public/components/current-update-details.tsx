import React from 'react';
import {
  EuiAccordion,
  EuiBadge,
  EuiCallOut,
  EuiFlexGroup,
  EuiFlexItem,
  EuiHeaderLink,
  EuiText,
  EuiSpacer,
} from '@elastic/eui';
import { Update } from '../../common/types';
import { FormattedMessage, I18nProvider } from '@osd/i18n/react';

export interface CurrentUpdateDetailsProps {
  currentUpdate?: Update;
}

export const CurrentUpdateDetails = ({ currentUpdate }: CurrentUpdateDetailsProps) => {
  if (!currentUpdate) {
    return null;
  }

  const currentRelease = `${currentUpdate?.semver.mayor}.${currentUpdate?.semver.minor}`;
  const releaseNotesUrl = `https://documentation.wazuh.com/${currentRelease}/release-notes/release-${currentUpdate.semver.mayor}-${currentUpdate.semver.minor}-${currentUpdate.semver.patch}.html`;
  const upgradeGuideUrl = `https://documentation.wazuh.com/${currentRelease}/upgrade-guide/index.html`;

  return (
    <I18nProvider>
      <EuiCallOut
        title={
          <EuiFlexGroup gutterSize="m">
            <EuiFlexItem grow={false} style={{ maxWidth: 'max-content' }}>
              <FormattedMessage
                id="wazuhCheckUpdates.currentUpdateDetails.title"
                defaultMessage="Wazuh new release is available now!"
              />
            </EuiFlexItem>
            <EuiFlexItem grow={false} style={{ maxWidth: 'max-content' }}>
              <EuiBadge>{currentUpdate.tag}</EuiBadge>
            </EuiFlexItem>
          </EuiFlexGroup>
        }
        color="warning"
        iconType="bell"
      >
        <EuiFlexGroup gutterSize="m">
          <EuiFlexItem grow={false} style={{ maxWidth: 'max-content' }}>
            <EuiHeaderLink
              href={releaseNotesUrl}
              isActive
              target="_blank"
              iconType="popout"
              iconSide="right"
            >
              <FormattedMessage
                id="wazuhCheckUpdates.currentUpdateDetails.releaseNotesLink"
                defaultMessage="Release notes"
              />
            </EuiHeaderLink>
          </EuiFlexItem>
          <EuiFlexItem grow={false} style={{ maxWidth: 'max-content' }}>
            <EuiHeaderLink
              href={upgradeGuideUrl}
              isActive
              target="_blank"
              iconType="popout"
              iconSide="right"
            >
              <FormattedMessage
                id="wazuhCheckUpdates.currentUpdateDetails.upgradeGuideLink"
                defaultMessage="Upgrade guide"
              />
            </EuiHeaderLink>
          </EuiFlexItem>
        </EuiFlexGroup>
        <EuiSpacer size="m" />
        <EuiAccordion
          id="accordionUpdateDetails"
          buttonContent={
            <FormattedMessage
              id="wazuhCheckUpdates.currentUpdateDetails.showDetails"
              defaultMessage="Show details"
            />
          }
          paddingSize="m"
        >
          <EuiText>
            {currentUpdate.description.split('\r\n').map((line, index) => (
              <p key={index}>{line}</p>
            ))}
          </EuiText>
        </EuiAccordion>
      </EuiCallOut>
    </I18nProvider>
  );
};
