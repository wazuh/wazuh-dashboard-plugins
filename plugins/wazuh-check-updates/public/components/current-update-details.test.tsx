import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { CurrentUpdateDetails } from './current-update-details';

jest.mock(
  '../../../../node_modules/@elastic/eui/lib/services/accessibility/html_id_generator',
  () => ({
    htmlIdGenerator: () => () => 'htmlId',
  })
);

describe('CurrentUpdateDetails component', () => {
  test('should render the current update tag and links to the Relese Notes and the Upgrade Guide', () => {
    const { container, getByText, getByRole } = render(
      // <TestProviders>
      <CurrentUpdateDetails
        currentUpdate={{
          title: 'Wazuh 4.2.6',
          description:
            'Wazuh 4.2.6 is now available. This version includes several bug fixes and improvements.',
          published_date: '2021-09-30T14:00:00.000Z',
          semver: {
            mayor: 4,
            minor: 2,
            patch: 6,
          },
          tag: '4.2.6',
        }}
      />
      // </TestProviders>
    );

    expect(container).toMatchSnapshot();

    const elementWithTag = getByText('4.2.6');
    expect(elementWithTag).toBeInTheDocument();

    const releaseNotesUrl = 'https://documentation.wazuh.com/4.2/release-notes/release-4-2-6.html';
    const releaseNotesLink = getByRole('link', { name: 'Release notes' });
    expect(releaseNotesLink).toHaveAttribute('href', releaseNotesUrl);

    const upgradeGuideUrl = `https://documentation.wazuh.com/4.2/upgrade-guide/index.html`;
    const upgradeGuideLink = getByRole('link', { name: 'Upgrade guide' });
    expect(upgradeGuideLink).toHaveAttribute('href', upgradeGuideUrl);
  });

  test('should return null when there is no current update', () => {
    const { container } = render(<CurrentUpdateDetails />);

    expect(container).toMatchSnapshot();

    const firstChild = container.firstChild;
    expect(firstChild).toBeNull();
  });
});
