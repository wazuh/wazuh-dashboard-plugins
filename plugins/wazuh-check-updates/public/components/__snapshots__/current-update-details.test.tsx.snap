// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentUpdateDetails component should render the current update tag and links to the Relese Notes and the Upgrade Guide 1`] = `
<div>
  <div
    class="euiCallOut euiCallOut--warning"
  >
    <div
      class="euiCallOutHeader"
    >
      <svg
        aria-hidden="true"
        class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiCallOutHeader__icon"
        focusable="false"
        height="16"
        role="img"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      />
      <span
        class="euiCallOutHeader__title"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="max-width: max-content;"
          >
            Wazuh new release is available now!
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="max-width: max-content;"
          >
            <span
              class="euiBadge euiBadge--iconLeft"
              style="background-color: rgb(211, 218, 230); color: rgb(0, 0, 0);"
              title="4.2.6"
            >
              <span
                class="euiBadge__content"
              >
                <span
                  class="euiBadge__text"
                >
                  4.2.6
                </span>
              </span>
            </span>
          </div>
        </div>
      </span>
    </div>
    <div
      class="euiText euiText--small"
    >
      <div
        class="euiTextColor euiTextColor--default"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterMedium euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="max-width: max-content;"
          >
            <a
              class="euiButtonEmpty euiButtonEmpty--primary euiHeaderLink euiHeaderLink-isActive"
              href="https://documentation.wazuh.com/4.2/release-notes/release-4-2-6.html"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
                <span
                  class="euiButtonEmpty__text"
                >
                  Release notes
                </span>
              </span>
            </a>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="max-width: max-content;"
          >
            <a
              class="euiButtonEmpty euiButtonEmpty--primary euiHeaderLink euiHeaderLink-isActive"
              href="https://documentation.wazuh.com/4.2/upgrade-guide/index.html"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
                <span
                  class="euiButtonEmpty__text"
                >
                  Upgrade guide
                </span>
              </span>
            </a>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--m"
        />
        <div
          class="euiAccordion"
        >
          <div
            class="euiAccordion__triggerWrapper"
          >
            <button
              aria-controls="accordionUpdateDetails"
              aria-expanded="false"
              class="euiAccordion__button"
              id="htmlId"
              type="button"
            >
              <span
                class="euiAccordion__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiIEFlexWrapFix"
              >
                Show details
              </span>
            </button>
          </div>
          <div
            aria-labelledby="htmlId"
            class="euiAccordion__childWrapper"
            id="accordionUpdateDetails"
            role="region"
            tabindex="-1"
          >
            <div>
              <div
                class="euiAccordion__padding--m"
              >
                <div
                  class="euiText euiText--medium"
                >
                  <p>
                    Wazuh 4.2.6 is now available. This version includes several bug fixes and improvements.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CurrentUpdateDetails component should return null when there is no current update 1`] = `<div />`;
